### PR TITLE
Adds Amazon Linux 2 def with apache hosted .NET 5 web applications in self-hosted and framework-dependent falvors

### PR DESCRIPTION
**Pull request by jaffinito**
- Adds Amazon Linux 2 def with ASP.NET Core and Apache
- Adds new generalized deploy-application section for Apache and RHEL (.net install auto-detects distro and does the right thing)
- deploy-application can deploy an app in a few different ways
  - The app can be self-hosted or framework-dependent
  - The app can be started as an executable or using dotnet the.dll
- Installs a systemd service from a generalized tempate based on def settings.
- Sets up apache as a reverse proxy for the app using def settings.
- Most things are configurable via the def file.
- Should support most distros based on rhel.

Note: The two new apps live in a different S3 than the other apps. They will need to be moved over at some point and the scripting updated.

### CODE REVIEW

File: test/manual/definitions/apm/.net/linux/linux2-aspnetcore5.json [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        "proxy_port": "5001"
+      }
+    },
+    {
+      "id": "dotnet2",
+      "destinations": [ "host1" ],
+      "source_repository": "https://github.com/newrelic/open-install-library.git",
+      "deploy_script_path": "test/deploy/linux/apache/deploy-application/.net/rhel/roles",
+      "port": 8081,
+      "proxy_port": "5000"
```
Comment (Julien4218) [inline test/manual/definitions/apm/.net/linux/linux2-aspnetcore5.json:49]: proxy_port should be within a "params" element

File: test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  when: use_executable|bool
+
+- name: Set startup_command to dotnet command (use dotnet xx.dll)
+  set_fact:
+    startup_command: "/usr/bin/dotnet /var/www/frameworkdependent-net5webapplication/net5webapplication.dll"
+  when: use_executable|bool == False
+
+- name: Set default download url for self contained web app
+  set_fact:
+    web_app_url: "https://virtuoso-testing.s3-us-west-2.amazonaws.com/selfcontained-net5webapplication.zip"
```
Comment (Julien4218) [inline test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml:32]: let's move this zip to the final S3 location to be consistent. cc @gabeobrien

File: test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  when: web_app_url is undefined and (is_selfcontained|bool)
+
+- name: Set default download url for framwork dependent web app
+  set_fact:
+    web_app_url: "https://virtuoso-testing.s3-us-west-2.amazonaws.com/frameworkdependent-net5webapplication.zip"
```
Comment (Julien4218) [inline test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml:37]: let's move this zip to the final S3 location to be consistent. cc @gabeobrien

**Review by Julien4218 — COMMENTED** [review]
@jaffinito this looks great. I've spotted a couple of things, and cc @gabeobrien

File: test/manual/definitions/apm/.net/linux/linux2-aspnetcore5.json (hunk 1 above)
Comment (jaffinito) [reply test/manual/definitions/apm/.net/linux/linux2-aspnetcore5.json]: fixed

File: test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml (hunk 2 above)
Comment (jaffinito) [reply test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml]: Worked with Gabe to get the apps in the right place.

File: test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml (hunk 3 above)
Comment (jaffinito) [reply test/deploy/linux/apache/deploy-application/.net/rhel/roles/onbeforestart/tasks/main.yml]: Worked with Gabe to get the apps in the right place.

**Comment by jaffinito** [comment]
@Julien4218 Fixed the proxy mistake and got the files in the right place.  I also updated the locations in the work Alex had done previously.

**Review by Julien4218 — APPROVED** [review]
This looks great, thank you @jaffinito

**Pull request merged**